pkg/retry/retry.go

```diff
@@ -36,3 +37,28 @@ func Run(initialInterval time.Duration, maxRetries uint64, f func() error) error
 		return err
 	}, retryCfg)
 }
+
+// RunWithInfiniteRetry retries the specified function indefinitely, until a backoff.PermanentError is encountered.
+// notifyFunc will be called each time before sleeping with the total elapsed time.
+func RunWithInfiniteRetry(initialInterval time.Duration, f func() error, notifyFunc func(elapsed time.Duration)) error {
+	cfg := backoff.NewExponentialBackOff()
+	cfg.InitialInterval = initialInterval
+	cfg.MaxElapsedTime = math.MaxInt64
+	cfg.MaxInterval = math.MaxInt64
```

Comment:
I prefer not to set the max interval for next retry (`MaxInterval`) too large, the default value `60s` may be enough